### PR TITLE
fix: [HL-30] Sign-In with Facebook✨

### LINKED ISSUE #60: fix issues inside sign-in-with -facebook

### PR DESCRIPTION
**Pull request by esraakhaled**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 1]
```diff
@@ -101,50 +105,68 @@ private extension LoginViewController {
...
-                // TODO: Show loading indicator.
-            }
-            .store(in: &subscriptions)
+//        viewModel.isLoadingIndicatorPublisher
+//            .sink { [weak self] _ in
+//                guard let _ = self else { return }
+//
+//                // TODO: Show loading indicator.
+//            }
+//            .store(in: &subscriptions)
```
Comment (oradwanomar) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]: remove commented code 
OR 
Add your loading indicator view 
Check this link that will help you make a simple, beautiful and reusable indicator view component 
https://youtu.be/iHeeOULNdFA

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 2]
```diff
@@ -101,50 +105,68 @@ private extension LoginViewController {
...
-                // TODO: Show error message.
-            }
-            .store(in: &subscriptions)
+//        viewModel.isShowErrorMessagePublisher
+//            .sink { [weak self] _ in
+//                guard let _ = self else { return }
+//
+//                // TODO: Show error message.
+//            }
+//            .store(in: &subscriptions)
```
Comment (oradwanomar) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]: Same as above

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 3]
```diff
@@ -101,50 +105,68 @@ private extension LoginViewController {
...
-                guard let _ = self else { return }
-                
-            }
-            .store(in: &subscriptions)
+//        viewModel.isLoginStatusPublisher
+//            .sink { [weak self] _ in
+//                guard let _ = self else { return }
+//
+//            }
+//            .store(in: &subscriptions)
```
Comment (oradwanomar) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]: same as above

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 4]
```diff
@@ -101,50 +105,68 @@ private extension LoginViewController {
...
+            } else {
+                debugPrint("Data: \(String(describing: credential))")
+            }
+        }
+    }
+
+    func loginButtonDidLogOut(_ loginButton: FBSDKLoginKit.FBLoginButton) {
+        let loginManager = LoginManager()
+        loginManager.logOut()
+        print("Logout")
```
Comment (oradwanomar) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]: delete any print statement in production

**Review by oradwanomar — CHANGES_REQUESTED** [review]

File: Healthy/Classes/AppDelegate.swift [hunk 5]
```diff
@@ -47,4 +62,10 @@ private extension AppDelegate {
         let config = GIDConfiguration(clientID: Constatns.googleClientId)
         GIDSignIn.sharedInstance.configuration = config
     }
+    func configureFacebookSignin(application: UIApplication,
```
Comment (ahmdmhasn) [inline Healthy/Classes/AppDelegate.swift:71]:
```suggestion
    }
    
    func configureFacebookSignin(application: UIApplication,
```

File: Healthy/Classes/AppDelegate.swift [hunk 6]
```diff
@@ -47,4 +62,10 @@ private extension AppDelegate {
...
+                                 launchOptions: [UIApplication.LaunchOptionsKey: Any]?) {
+        ApplicationDelegate.shared.application(
+            application,
+            didFinishLaunchingWithOptions: launchOptions)
```
Comment (ahmdmhasn) [inline Healthy/Classes/AppDelegate.swift]:
```suggestion
            didFinishLaunchingWithOptions: launchOptions
        )
```

File: Healthy/Classes/Modules/Login/LoginViewModel.swift [hunk 7]
```diff
@@ -35,17 +35,16 @@ extension LoginViewModel: LoginViewModelOutput {
     var isLoadingIndicatorPublisher: AnyPublisher<Bool, Never> {
         $isLoadingState.eraseToAnyPublisher()
     }
-    
+
     var isShowErrorMessagePublisher: AnyPublisher<String, Never> {
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Login/LoginViewModel.swift]:
```suggestion
    var errorMessagePublisher: AnyPublisher<String, Never> {
```

is does reflect a boolean not a noun. If this is out of the scope, please add todo.

File: HealthyTests/Classes/Modules/Login/Mocks/LoginViewModelMock.swift [hunk 8]
```diff
@@ -1,6 +1,32 @@
...
+        $email
+            .map { $0.isEmpty }
+            .eraseToAnyPublisher()
+    }
+
+    var isLoginStatusPublisher: AnyPublisher<Bool, Never> {
+        $email
+            .map { $0.isEmpty }
+            .eraseToAnyPublisher()
+    }
```
Comment (ahmdmhasn) [inline HealthyTests/Classes/Modules/Login/Mocks/LoginViewModelMock.swift]:
```suggestion
    var isLoadingIndicatorPublisher: AnyPublisher<Bool, Never> {
        Just(false).eraseToAnyPublisher()
    }

    var isShowErrorMessagePublisher: AnyPublisher<String, Never> {
        Just("").eraseToAnyPublisher()
    }

    var isLoginEnabledPublisher: AnyPublisher<Bool, Never> {
        Just(false).eraseToAnyPublisher()
    }

    var isLoginStatusPublisher: AnyPublisher<Bool, Never> {
        Just(false).eraseToAnyPublisher()
    }
```

File: Healthy/Classes/Utilities/Authentication/Login/FacebookAuthenticator.swift [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import Foundation
+import UIKit
+import FBSDKLoginKit
+
+// Provide AppID
+class FacebookAuthenticator: NSObject {
```
Comment (ahmdmhasn) [inline Healthy/Classes/Utilities/Authentication/Login/FacebookAuthenticator.swift:6]: You need to conform authenticator and move all the needed logic inside authenticate. Check Google authenticator please.

**Comment by ahmdmhasn** [comment]
@esraakhaled Please re-request review when the changes are ready.

**Comment by esraakhaled** [comment]
it reports a conflict when merging develop

**Pull request closed without merging**